docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+
+
+The reserved generator name for such templates is `dm_template`.
+
+#### Go Templates (formerly Helm Templates)
+Helm 0.3 and later supported Go templates, [defined here](https://github.com/helm/helm/blob/master/docs/generate-and-template.md). The only change this proposal makes to the existing specification is that templates must include the form of the `helm:generate` header elaborated here.
+
+
+[https://github.com/helm/helm/blob/master/docs/generate-and-template.md]
+
+
+The reserved generator name for such templates is `go_template`.
+
+
+Example:
+
+
+The following simple manifest (`manifests/secret.yaml`) declares a secret that is randomly generated. The Go template directive is bolded.
+
+
+```
+#helm:generate go_template
+apiVersion: v1
+kind: Secret
+metadata:
+ name: builder-key-auth
+ namespace: deis
+ labels:
+   heritage: deis
+type: Opaque
+data:
+ builder-key: {{ randAscii 64 | b64enc }}
+```
+
+The above generates a 64 character random ASCII string and then base-64 encodes it.
+
+
+### The Chart File
+The `Chart.yaml` file specifies package metadata for the definition. Package metadata is any information that explains the package structure, purpose, and provenance. Its intended audience is tooling that surfaces such information to the user (e.g. a command line client, a web interface, a search engine).
+
+
+A definition file does not specify anything about individual pieces of the definition (e.g. description of per-field schema or metadata), nor does it contain information about the parent environment (e.g. that it is hosted on S3 or in GitHub). Its scope is the definition as a whole.
+
+
+Fields:
+
+
+* Name: A human-readable name of the definition. This may contain UTF-8 alphanumeric text, together with dash and underscore characters.
+* Description: A human-readable description of the definition, not to exceed one paragraph of text. No text formatting is supported. A description may use any printable text characters allowed by the file format.
+* Version: A SemVer 2 semantic version of the chart (template files). While the chart may reflect the version of the software it describes, this is not necessary. However, to follow semantic versioning constraints, every revision of the chart MUST increment the version in a way that indicates incremental change (e.g. simply changing the build ID is insufficient, as by definition it does not indicate incremental change over another build ID).
+* Keywords: A list of human-readable keywords. A keyword may contain alphanumeric text and spaces.
+* Maintainers: A list of author objects, where an author object has two properties:
+   ** Name: Author name
+   ** Email: Author email (optional)
+* Source: A URL to the source code for this chart
+* Home: A URL to the home page for the project
+* Dependencies:
```

Comment:
Yes, those are complex use cases that aren't fully addressed by most tools, including production DM. Makes sense to handle them with a separate design document, as you suggest.

I went ahead and merged the document to move things along, and to keep the list of changes in each PR manageable. We should, however, clean up the text to say that the dependencies are still under development, and point to the new design document. You can go ahead and submit another PR for that change whenever you're ready.